server/src/test/java/org/elasticsearch/common/joda/JavaJodaTimeDuellingTests.java

```diff
@@ -390,6 +391,28 @@ public void testDuelingStrictParsing() {
 
         assertSameDate("2012-W31-5", "strict_weekyear_week_day");
         assertParseException("2012-W1-1", "strict_weekyear_week_day");
+//failing below
+        assertSameDate("2015-01-04T00:00Z", "strict_date_optional_time||epoch_millis");
+    }
+//Failing
```

Comment:
will delete these if the PR is ok, but it helps investigating

Reply:
for sure I'm fine with the PR in general! :-)